src/log.py

```diff
@@ -0,0 +1,107 @@
+"""
+log
+ 
+By: Norma Chang
+"""
+import json
+import datetime
+import time
+ 
+ 
+class Log:
+    """
+    Log class.
+ 
+    Generates a log file that contains the data from the properties of an instance of the leader/follower class.
+ 
+    Array of dictionaries is generated when the log file is loaded.
+    """
+    def __init__(self):
+        self._start_time = str(datetime.datetime.now())
+
+
+    def log_to_file(self, robot_obj, role):
+        """
+        Logs data from leader or follower.
+ 
+        The roles are given in the when the function is called, if the roles are given in the proper format
+        the script will log the data from the properties of that class instance into dictionaries. If the role is not an accepted role or not in the correct format, an exception is raised.
+ 
+        :param robot_obj: Where the instance of the class is passed in. 'follower' for the follower class instance and 'leader' for the leader class instance.
+        :type robot_obj: object
+ 
+        :param role: Role of the class instance. 'follower' for the follower role and 'leader' for the leader role.
+        :type: string
+        """
+        if role == 'leader':
+            leader_dict = {
+                'Time' : time.time(),
+                'Input' : robot_obj._distance,
+                'Speed' : robot_obj._speed,
+            }
+            self.write_to_file(leader_dict, role)
+        elif role == 'follower':
+            follower_dict = {
+                'Time' : time.time(),
+                'Distance' : robot_obj._distance,
+                'Speed' : robot_obj._speed,
+                'Turn Angle' : robot_obj._turn_angle,
+                'Yaw' : robot_obj._yaw,
+            }
+            self.write_to_file(follower_dict, role)
+        else:
+            # Print error message if role is not accepted or in proper format.
+            raise Exception('Role must be either a leader or follower. The role was: {}'.format(role))
```

Comment:
This is too finicky. I don't like the string idea. Can you find a better way to check the role?

Reply:
I will look into it

Reply:
Is there a certain way you envision checking the role? I could check to see what vehicle is running.